### PR TITLE
ci: Federation auth method for gcp registry

### PR DESCRIPTION
**Pull request by alpeto9**
## Description

Login to gpc registry through federations instead of long live tokens

### CODE REVIEW

File: .github/workflows/docker.yml [hunk 1]
```diff
@@ -87,12 +99,14 @@ jobs:
...
-        uses: docker/login-action@v1
+      - id: auth
+        uses: google-github-actions/auth@a61909d048e0be579b6c15b27088d19668493851
         with:
-          registry: eu.gcr.io
-          username: _json_key
-          password: ${{ secrets.GCR_EU_DEV_JSON_KEY }}
+          workload_identity_provider: ${{ secrets.GCP_WORKLOAD_IDP }}
+          service_account: ${{ secrets.GCP_SERVICE_ACCOUNT }}
+      - name: configure-docker
```
Comment (dhedey) [inline .github/workflows/docker.yml:107]: Can we give this a nicer name?

File: .github/workflows/docker.yml [hunk 2]
```diff
@@ -54,12 +58,16 @@ jobs:
...
+        uses: google-github-actions/auth@a61909d048e0be579b6c15b27088d19668493851
         with:
-          registry: eu.gcr.io
-          username: _json_key
-          password: ${{ secrets.GCR_EU_DEV_JSON_KEY }}
+          workload_identity_provider: ${{ secrets.GCP_WORKLOAD_IDP }}
+          service_account: ${{ secrets.GCP_SERVICE_ACCOUNT }}
+      - name: 'Set up Cloud SDK'
+        uses: 'google-github-actions/setup-gcloud@v1'
+      - name: configure-docker
```
Comment (dhedey) [inline .github/workflows/docker.yml:68]: Can we give this a nicer name?

**Review by dhedey — COMMENTED** [review]
Looks good - a couple of comments. Will let devops review.

Can you update the PR description (perhaps remove it if it's not relevant?)

**Review by dhedey — APPROVED** [review]
Approving from Network PoV - but I'm not really sure what this changes / why - perhaps a slack post might help?

**Review by shambupujar — APPROVED** [review]

**Pull request merged**